pool/src/lib.rs

```diff
@@ -539,10 +540,18 @@ impl LendingPoolTrait for LendingPool {
             mint: false,
             who: who.clone(),
         };
-        add_stoken_underlying_balance(&env, &reserve.s_token_address, amount)?;
+        let mint_burn_2 = MintBurn {
```

Comment:
Looks like internal balance can go to negative. Suggest to add method with setting balance and make non negative checks in add_token_balance & add_token_total_supply

Reply:
Also need to add one more add_token_balance, for mint_burn_1